graphrag/index/flows/create_final_embeddings.py

```diff
@@ -0,0 +1,211 @@
+                filename="create_final_entities_description_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Entities embeddings (description)",
+            )
+
+        # Embed title in community reports table
+        if community_title_embedding in embedded_fields and isinstance(
+            community_reports_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=community_reports_embeddings.copy(),
+                column_to_embed="title",
+                filename="create_final_community_reports_title_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Community reports embeddings (title)",
+            )
+
+        # Embed summary in community reports table
+        if community_summary_embedding in embedded_fields and isinstance(
+            community_reports_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=community_reports_embeddings.copy(),
+                column_to_embed="summary",
+                filename="create_final_community_reports_summary_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Community reports embeddings (summary)",
+            )
+
+        # Embed full_content in community reports table
+        if community_full_content_embedding in embedded_fields and isinstance(
+            community_reports_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=community_reports_embeddings.copy(),
+                column_to_embed="full_content",
+                filename="create_final_community_reports_full_content_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Community reports embeddings (full_content)",
+            )
+
+    return pd.DataFrame()
+
+
+async def run_embeddings(
+    data: pd.DataFrame,
+    column_to_embed: str,
+    filename: str,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    storage: PipelineStorage,
+    base_text_embed: dict | None = None,
+    message: str | None = None,
+) -> None:
+    """All the steps to generate single embedding."""
+    if base_text_embed:
+        log.info("%s", message)
+        data["embedding"] = await embed_text(
+            data,
+            callbacks,
+            cache,
+            column=column_to_embed,
+            strategy=base_text_embed["strategy"],
+        )
+
+        data = cast(pd.DataFrame, data[["id", "embedding"]])
```

Comment:
I think this cast isn't necessary

Reply:
it was requested by ruff in the format check

Reply:
you can use methods than ensure data frame retrieval. For example, instead of df["val] do df.loc[:, "val"]